File: api/src/Domain/ProductCatalog/DataTransferObjects/PadColorCreateOrUpdate.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DetailingArsenal.Domain.ProductCatalog;
using DetailingArsenal.Domain.Shared;

namespace DetailingArsenal.Domain.ProductCatalog {
    public class PadColorCreateOrUpdate : IDataTransferObject {
        public Guid? Id { get; }
        public string Name { get; }
        public PadCategory Category { get; }
        public DataUrlImage? Image { get; }
        public List<PadOptionCreateOrUpdate> Options { get; }

        public PadColorCreateOrUpdate(Guid? id, string name, PadCategory category, DataUrlImage? image, List<PadOptionCreateOrUpdate> options) {
            Id = id;
            Name = name;
            Category = category;
            Image = image;
            Options = options;
        }
    }
}

File: app/src/api/product-catalog/data-transfer-objects/requests/pad-color-create-or-update.ts
import { Image } from "../image";
import { PadOption } from "../pad-option";
import { PadOptionCreateOrUpdate } from "./pad-option-create-or-update";

export interface PadColorCreateOrUpdate {
    id: string | null,
    name: string,
    category: string,
    image: Image | string | null,
    options: PadOptionCreateOrUpdate[]
}